osde2e/cloud_ingress_operator_tests.go

```diff
@@ -138,6 +129,21 @@ var _ = ginkgo.Describe("cloud-ingress-operator", ginkgo.Ordered, func() {
 		// Make sure both the New cidrBlock and the Service LoadBalancerSourceRanges are equal
 		// If they are then the APIScheme update also updated the service.
 		Expect(updatedCidrBlock).To(BeEquivalentTo(rhAPIService.Spec.LoadBalancerSourceRanges), "Updated cidrblock from apischeme did not reflect in rh-api service")
+
+		// Finally, restore the CIDR blocks back to the original state.
+		updatedApiScheme.Spec.ManagementAPIServerIngress.AllowedCIDRBlocks = apiScheme.Spec.ManagementAPIServerIngress.AllowedCIDRBlocks
+		err = k8s.Update(ctx, updatedApiScheme)
+		Expect(err).NotTo(HaveOccurred(), "Could not revert APIScheme CR instance")
+
+		ginkgo.By("Restoring the CIDR block to its original state.")
+		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 5*time.Minute, false, func(ctx context.Context) (bool, error) {
+			if slices.Equal(updatedApiScheme.Spec.ManagementAPIServerIngress.AllowedCIDRBlocks, cidrBlock) {
+				log.Println("Original cidrblock from apischeme successfully restored in rh-api service.")
+				return true, nil
+			}
+			return false, nil
+		})
+		Expect(err).NotTo(HaveOccurred(), "Original cidrblock not restored in APIScheme.")
```

Comment:
This is the fix for the unit test.  Blocking until the state has been restored improves the stability of the test suite.

Reply:
It's not a 5 min timeout, that's the deadline timeout, so if it doesn't succeed within 5 mins it will fail.  It usually succeeds < 30s.